### PR TITLE
B 21533 int HQ role with Safety Move privs

### PR DESCRIPTION
**Pull request by r-mettler**
## Summary

Previously code was added to prevent HQ role from being able to have Safety Move privileges, but it was decided that HQ should now have access to see Safety Moves. This PR removes the checks preventing the HQ role from selecting the Safety Move priv in the admin console.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

To be able to select Safety Moves you need to first add the Super permissions for the admin account. This can be done by changing the flag in the DB for `super` to true in the `admin_users` table.

image

Now you can either edit and existing Office User and give them Headquarters Role and Safety Move privileges or create a new Office User with the HQ role and Safety Move priv. 

NOTE: If creating a new Office User you first must select the HQ role and save and then you can add the Safety Move privilege. This is because the checkbox for Safety Move is only currently available on the edit Office User page and not create page (known issue). 

1. With an admin who has super powers (DB flag see above) access the admin app
2. Create or Edit an existing Office User and give them at least the Service Counselor and Headquarters Role and Safety Move privileges. Make sure to remember which GBLOC you gave the Office User.
3. While viewing the create or edit Office User page you should see the following label:
### The Safety Moves privilege can only be selected for the following roles: Task Ordering Officer, Task Invoicing Officer, Services Counselor, Quality Assurance Evaluator, Customer Service Representative, and Headquarters
4. Verify you can select Safety Moves privileges with the following roles and any combination of: 

Task Ordering Officer
Task Invoicing Officer
Services Counselor
Quality Assurance Evaluator
Customer Service Representative
Headquarters

5. Verify you can NOT select Safety Move privileges with the following roles and any combination of:

Customer
Contracting Officer
Prime Simulator
Government Surveillance Representative

6. Login to the Office App as the Service Counselor, go to the Customer Search and type in any and perform the search
7. Click the Add Customer button
8. When creating the customer you will be asked if this is a Safety Move, select Yes
9. Finish filling out the fields and create a new move
10. Go through and create an HHG shipment, make sure you set it up so that your shipment falls under the same GBLOC as your Office User so you will see the move in your Queue
11. Make sure you enter all the Order information but don't submit the shipment yet, copy your move code
12. Change your user role to Headquarters and go to the Counseling Queue
13. Verify you can see your Safety Move (should have a ${\color{red}Safety}$ label and move code starts with `SM`)
14. Select the Safety Move and verify you can view the shipment
15. Switch the role back to Service Counselor and Submit the shipment
16. Change back to the Headquarters role and verify you can see the shipment in the  TOO queue
17. Select the shipment and verify you can view it

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### CODE REVIEW

**Review by ajlusk — APPROVED** [review]
I was unable to apply the Safety Moves privilege when Customer, Prime Sim, Contracting Officer, or GSR were selected. I was unable to apply the Customer, Prime Sim, Contracting Officer, or GSR roles when Safety Moves were selected. Encountered no limitations with other combinations in either order for that privilege.

Code changes look good after a couple some comments outside of Github.

In the next BL for this feature it might be good to add a playwright flow to check these limitations since we can't test the admin app with normal React tests.

**Review by MInthavongsay — APPROVED** [review]
Looks good. Could approve on the styling of the SAFETY label in the details page and listing; but, thats outside the scope of this BL.  See screenshot of possible styling enhancement for the label.
Screenshot 2024-12-11 at 3 42 10 PM

**Comment by r-mettler** [comment]
> Looks good. Could improve on the styling of the SAFETY label in the details page and listing; but, thats outside the scope of this BL. See screenshot of possible styling enhancement for the label. Screenshot 2024-12-11 at 3 42 10 PM

I like that a LOT better, can run it by Wendy and Dre and maybe just slip in the change somewhere 😄

**Comment by danieljordan-caci** [comment]
> > Looks good. Could improve on the styling of the SAFETY label in the details page and listing; but, thats outside the scope of this BL. See screenshot of possible styling enhancement for the label. Screenshot 2024-12-11 at 3 42 10 PM
> 
> I like that a LOT better, can run it by Wendy and Dre and maybe just slip in the change somewhere 😄

100% recommend using what Mike provided. Looks a lot better. I actually have a feature in place to handle fixing the CANCELED and BLUEBARK tags with `ShipmentTag`

**Comment by r-mettler** [comment]
> I was unable to apply the Safety Moves privilege when Customer, Prime Sim, Contracting Officer, or GSR were selected. I was unable to apply the Customer, Prime Sim, Contracting Officer, or GSR roles when Safety Moves were selected. Encountered no limitations with other combinations in either order for that privilege.
> 
> Code changes look good after a couple comments outside of Github.
> 
> In the next BL for this feature it might be good to add a playwright flow to check these limitations since we can't test the admin app with normal React tests.

I looked into this some and due to the lack of being able to actually set an admin user to "Super admin" via the admin app I am not sure how we would set it via DB when an admin account is created to do the playwright tests. The Safety Moves priv checkbox will not show up to test unless you are an admin with super privileges. We certainly can easily check the Supervisor priv via playwright tests in the downstream.

**Comment by danieljordan-caci** [comment]
> > I was unable to apply the Safety Moves privilege when Customer, Prime Sim, Contracting Officer, or GSR were selected. I was unable to apply the Customer, Prime Sim, Contracting Officer, or GSR roles when Safety Moves were selected. Encountered no limitations with other combinations in either order for that privilege.
> > Code changes look good after a couple comments outside of Github.
> > In the next BL for this feature it might be good to add a playwright flow to check these limitations since we can't test the admin app with normal React tests.
> 
> I looked into this some and due to the lack of being able to actually set an admin user to "Super admin" via the admin app I am not sure how we would set it via DB when an admin account is created to do the playwright tests. The Safety Moves priv checkbox will not show up to test unless you are an admin with super privileges. We certainly can easily check the Supervisor priv via playwright tests in the downstream.

You can create an admin user via factory with Super privileges prior to running the test?

**Comment by r-mettler** [comment]
> > > I was unable to apply the Safety Moves privilege when Customer, Prime Sim, Contracting Officer, or GSR were selected. I was unable to apply the Customer, Prime Sim, Contracting Officer, or GSR roles when Safety Moves were selected. Encountered no limitations with other combinations in either order for that privilege.
> > > Code changes look good after a couple comments outside of Github.
> > > In the next BL for this feature it might be good to add a playwright flow to check these limitations since we can't test the admin app with normal React tests.
> > 
> > 
> > I looked into this some and due to the lack of being able to actually set an admin user to "Super admin" via the admin app I am not sure how we would set it via DB when an admin account is created to do the playwright tests. The Safety Moves priv checkbox will not show up to test unless you are an admin with super privileges. We certainly can easily check the Supervisor priv via playwright tests in the downstream.
> 
> You can create an admin user via factory with Super privileges prior to running the test?

I figured out a way to add in the creation of a user with super privileges and added a few tests.

**Review by antgmann — APPROVED** [review]
Testing steps work nicely, though I keep forgetting office users need to manually change gblocs lol. New tests look great, looking forward to leveraging some of this in the downstream.

**Review by paulstonebraker — APPROVED** [review]
Everything works and I learned a bunch about playwright (namely that I need to be updating playwright tests) during review! Also I was reminded how much i hate `reduce`

File: src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx [hunk 1]
```diff
@@ -36,11 +36,8 @@ const RolesPrivilegesCheckboxInput = (props) => {
   };
 
   const parseRolesCheckboxInput = (input) => {
-    if (
-      privilegesSelected.includes(elevatedPrivilegeTypes.SUPERVISOR) ||
-      privilegesSelected.includes(elevatedPrivilegeTypes.SAFETY)
-    ) {
-      var index;
+    var index;
```
[the remark this answers is not in the document]
Comment (r-mettler) [reply src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx]: This use case was me just moving what was there before outside of the if statement. You are right this should be `let` I went ahead and changed it.

**Review by antgmann — APPROVED** [review]

File: src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx [hunk 2]
```diff
@@ -93,24 +110,21 @@ const RolesPrivilegesCheckboxInput = (props) => {
   };
 
   const parsePrivilegesCheckboxInput = (input) => {
+    var index;
```
Comment (ajlusk) [inline src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx:113]: Looks like you missed a `var` though

**Review by ajlusk — APPROVED** [review]
Looks good! One non-blocking note. Appreciate the helpful but not strictly necessary test data updates.

File: src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx (hunk 2 above)
Comment (r-mettler) [reply src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx:113]: Having Anthony fix that in his downstream PR

**Pull request merged**